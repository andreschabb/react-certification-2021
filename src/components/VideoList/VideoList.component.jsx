import React from 'react';

import VideoItem from '../VideoItem';
import { GridContainer } from './VideoList.styles';

const VideoList = ({ videos }) => {
  const renderedVideos = videos.map((video) => {
    return <VideoItem key={video.id.videoId} video={video} videos={videos} />;
  });

  return <GridContainer data-testid="video-list">{renderedVideos}</GridContainer>;
};

export default VideoList;
