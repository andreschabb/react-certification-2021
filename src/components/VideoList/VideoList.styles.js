import styled from 'styled-components';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: 25% 25% 25% 25%;

  @media screen and (max-width: 1000px) {
    grid-template-columns: 33% 33% 33%;
  }

  @media screen and (max-width: 750px) {
    grid-template-columns: 50% 50%;
  }

  @media screen and (max-width: 500px) {
    grid-template-columns: 100%;
  }
`;
