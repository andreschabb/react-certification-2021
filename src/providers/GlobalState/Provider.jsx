import React, { createContext, useContext, useReducer } from 'react';

import reducer from './Reducer';

const initState = {
  isThemeLight: true,
};

const GlobalStateContext = createContext({
  isThemeLight: true,
});

function useGlobalState() {
  const context = useContext(GlobalStateContext);
  if (!context) {
    throw new Error(`Can't use "useTodos" without an TodoProvider!`);
  }
  return context;
}

function GloablStateProvider({ children }) {
  const [state, dispatch] = useReducer(reducer, initState);

  return (
    <GlobalStateContext.Provider value={{ state, dispatch }}>
      {children}
    </GlobalStateContext.Provider>
  );
}

export { useGlobalState };

export default GloablStateProvider;
