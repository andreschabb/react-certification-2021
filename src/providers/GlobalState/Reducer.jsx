export default function reducer(state, action) {
  switch (action.type) {
    case 'CHANGE_THEME': {
      return {
        ...state,
        isThemeLight: action.payload,
      };
    }
    default:
      throw new Error('Unkown action');
  }
}
