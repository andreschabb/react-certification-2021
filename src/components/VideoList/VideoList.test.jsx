import React from 'react';
import { render, screen } from '@testing-library/react';

import VideoList from './VideoList.component';
import { useGlobalState } from '../../providers/GlobalState/Provider';

const mockedVideos = [
  {
    id: { kind: 'youtube#video', videoId: 'nmXMgqjQzls' },
    snippet: {
      title: 'Video Tour | Welcome to Wizeline Guadalajara',
      description:
        'Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...',
      thumbnails: {
        medium: {
          url: 'https://i.ytimg.com/vi/nmXMgqjQzls/medium.jpg',
          width: 120,
          height: 90,
        },
      },
    },
  },
  {
    id: { kind: 'youtube#video', videoId: 'nmXMgqjQlhj' },
    snippet: {
      title: 'Title',
      description: 'Description',
      thumbnails: {
        medium: {
          url: 'https://i.ytimg.com/vi/nmXMgqjQzls/medium.jpg',
          width: 120,
          height: 90,
        },
      },
    },
  },
];

jest.mock('../../providers/GlobalState/Provider');

describe('VideoList', () => {
  beforeEach(() => {
    useGlobalState.mockImplementation(() => ({
      state: { isThemeLight: true },
    }));
  });

  it('should renders', () => {
    const { container } = render(<VideoList videos={mockedVideos} />);

    expect(screen.getByTestId('video-list')).toBeTruthy();

    expect(container).toMatchSnapshot();
  });

  it('should renders two video items', () => {
    render(<VideoList videos={mockedVideos} />);

    expect(screen.getAllByTestId('video-item').length).toBe(2);
  });
});
