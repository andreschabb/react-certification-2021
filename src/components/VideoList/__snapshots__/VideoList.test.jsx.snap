// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList should renders 1`] = `
<div>
  <div
    class="sc-gKsewC ikKpsV"
    data-testid="video-list"
  >
    <div
      class="sc-bdfBwQ jaBdCV"
      data-testid="video-item"
    >
      <div
        class="sc-gsTCUz krmAWT"
      >
        <img
          alt="Video Tour | Welcome to Wizeline Guadalajara"
          class="sc-dlfnbm isGCyU"
          src="https://i.ytimg.com/vi/nmXMgqjQzls/medium.jpg"
        />
        <div
          class="sc-hKgILt geQTeo"
        >
          <h4
            class="sc-eCssSg jMQuYw"
          >
            Video Tour | Welcome to Wizeline Guadalajara
          </h4>
          <p
            class="sc-jSgupP hWSoYQ"
          >
            Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
          </p>
        </div>
      </div>
    </div>
    <div
      class="sc-bdfBwQ jaBdCV"
      data-testid="video-item"
    >
      <div
        class="sc-gsTCUz krmAWT"
      >
        <img
          alt="Title"
          class="sc-dlfnbm isGCyU"
          src="https://i.ytimg.com/vi/nmXMgqjQzls/medium.jpg"
        />
        <div
          class="sc-hKgILt geQTeo"
        >
          <h4
            class="sc-eCssSg jMQuYw"
          >
            Title
          </h4>
          <p
            class="sc-jSgupP hWSoYQ"
          >
            Description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
